package/scripts/build/dawn.ts

```diff
@@ -24,6 +24,8 @@ const PLATFORM_MAP: Record<string, string> = {
   arm64_iphoneos: "OS64",
   arm64_iphonesimulator: "SIMULATORARM64",
   x86_64_iphonesimulator: "SIMULATOR64",
+  arm64_xros: "VISIONOS",
+  arm64_xrsimulator: "SIMULATOR_VISIONOS",
```

Comment:
there is no x86_64_xrsimulator right?

Reply:
This is tricky because there shouldn't be x86 simulator (because x86 macs are not supported for visionOS but I think we might need also to build it for x86 and include it in the slice). 

When working with Hermes prebuilt, I encountered some issues when trying to upload the app to the App Store.  (https://github.com/facebook/react-native/pull/45911)

However at the moment there is no x86 build option in the `ios.toolchain.cmake`. Im going to add it and then contribute this back to https://github.com/leetal/ios-cmake

Reply:
I am building it now, next should I just try to run the example app on the xrsimulator?

Reply:
Uhm, its not so simple unfortunately :D We would need to setup visionOS in the example app: https://callstack.github.io/react-native-visionos-docs/docs/getting-started/integrate-into-existing-projects 

Or create a new example. 

At this point visionOS doesn't support skia so we might need to also disable it for visionOS. Now that I got the build working Im going to try to integrate visionOS into the example and run it there

Reply:
I've integrated visionOS into the example app and it (almost) works, for some reason the `RectButton` was not working so I swapped it for TouchableOpacity and it works! 

Are you okay with keeping visionOS as an additional platform in the example app? (I can assist you with any updates / fixes regarding this platform)